### PR TITLE
DL-6098, generalise authentication and add all roles restriction to receipt of timescales data

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/controllers/actions/PrivilegedAction.scala [hunk 1]
```diff
@@ -16,64 +16,55 @@
...
 import models.requests.IdentifierRequest
 import play.api.mvc.Results._
 import play.api.mvc._
-import play.api.{Configuration, Environment, Logger}
-import uk.gov.hmrc.auth.core.AuthProvider.PrivilegedApplication
+import play.api.Logger
+import uk.gov.hmrc.auth.core.authorise.Predicate
 import uk.gov.hmrc.auth.core._
-import uk.gov.hmrc.auth.core.retrieve.{Credentials, Name, ~}
+import uk.gov.hmrc.auth.core.authorise.Predicate
```
Comment (paulanderson1234) [inline app/controllers/actions/PrivilegedAction.scala:25]: Duplicate ?

File: app/controllers/TwoEyeReviewController.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ class TwoEyeReviewController @Inject() (
 
   val logger = Logger(getClass)
```
Comment (paulanderson1234) [inline app/controllers/TwoEyeReviewController.scala:43]: Couple of these where type annotation is requested in IntelliJ - no problem if we do not want to supply it - just an observation

File: app/controllers/actions/PrivilegedAction.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/controllers/actions/PrivilegedAction.scala:25]: yes, the first only imports things immediately in core, so things under retrieve and authorise need to be done separately

File: test/controllers/actions/AuthenticatedIdentifierActionSpec.scala [hunk 3]
```diff
@@ -33,7 +33,7 @@ import scala.concurrent.{ExecutionContext, Future}
 class AuthenticatedIdentifierActionSpec extends ControllerBaseSpec with MockAuthConnector {
```
Comment (paulanderson1234) [inline test/controllers/actions/AuthenticatedIdentifierActionSpec.scala:33]: Just questioning why we kept this name if we changed the main controller to AllRolesAuthenticatedAction ?

File: app/controllers/TwoEyeReviewController.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/controllers/TwoEyeReviewController.scala]: added now

File: test/controllers/actions/FactCheckerAuthenticatedIdentifierActionSpec.scala [hunk 4]
```diff
@@ -33,7 +33,7 @@ import scala.concurrent.{ExecutionContext, Future}
 class FactCheckerAuthenticatedIdentifierActionSpec extends ControllerBaseSpec with MockAuthConnector {
```
Comment (paulanderson1234) [inline test/controllers/actions/FactCheckerAuthenticatedIdentifierActionSpec.scala:33]: Same as above if we changed the main controller name then perhaps we could have changed the test Spec name ?

File: test/controllers/actions/AuthenticatedIdentifierActionSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/controllers/actions/AuthenticatedIdentifierActionSpec.scala:33]: Missed them, will do now

File: test/controllers/actions/FactCheckerAuthenticatedIdentifierActionSpec.scala (hunk 4 above)
Comment (cmrhmrc) [reply test/controllers/actions/FactCheckerAuthenticatedIdentifierActionSpec.scala:33]: As above

Comment (cmrhmrc) [reply test/controllers/actions/FactCheckerAuthenticatedIdentifierActionSpec.scala:33]: Found a few more in same package, updating

File: app/controllers/actions/PrivilegedAction.scala (hunk 1 above)
Comment (paulanderson1234) [reply app/controllers/actions/PrivilegedAction.scala:24]: Optimised - resolved

**Pull request merged**